spec/errors.md

```diff
@@ -16,10 +16,10 @@ and MUST be emitted as soon as possible.
 The other error categories are only emitted during formatting,
 but it might be possible to detect them with validation tools.
 
-During selection, an _expression_ handler MUST only emit _Resolution Errors_ and _Selection Errors_.
-During formatting, an _expression_ handler MUST only emit _Resolution Errors_ and _Formatting Errors_.
+During selection and formatting,
+_expression_ handlers MUST only emit _Message Function Errors_.
 
-_Resolution Errors_ and _Formatting Errors_ in _expressions_ that are not used
+_Resolution Errors_ and _Message Function Errors_ in _expressions_ that are not used
 in _pattern selection_ or _formatting_ MAY be ignored,
 as they do not affect the output of the formatter.
```

Comment:
Thought: can a _Message Function Error_ ("MFE") occur in an unused _expression_? You can't have an error from a function that is never called. Admittedly, some of the MFEs might be discerned from the registry, e.g. checking option names or option values, during parsing. Again, maybe turn this upside down:

```suggestion
Implementations do not have to check for or emit _Resolution Errors_
or _Message Function Errors_ in _expressions_ that are not otherwise used by the _message_,
such as _placeholders_ in unselected _patterns_
or _declarations_ that are never referenced during _formatting_.
```